lib/shift.rb

```diff
@@ -1,21 +1,25 @@
 class Shift
 
-    def initialize(key, date, direction = :encrypt, set_size = nil)
-      set_size = CharacterSet.new.default_set.size if !set_size
-      date_squared = date.to_i ** 2
-
+    def initialize(key, date, direction = :encrypt)
+      @key  = key
+      @date = date
       @offsets = [
-                    offset_a(key, date_squared),
-                    offset_b(key, date_squared),
-                    offset_c(key, date_squared),
-                    offset_d(key, date_squared)
+                    shift_a,
+                    shift_b,
+                    shift_c,
+                    shift_d
                   ]
       set_direction(direction)
+      # All theses methods and typing, just to do this...
       # 4.times do |i|
       #   @offsets << key[i..i+1].to_i + date_squared / 10 ** (3 - i) % 10
       # end
```

Comment:
Remember to ask instructors if this would be acceptable instead of all the methods